recite/admin.py

```diff
@@ -55,89 +69,91 @@ class ReciterAdmin(admin.ModelAdmin):
-                    segments=segments,
-                    audio=file_ayah
-                )
-
-        shutil.rmtree(temp_dir)
+        audio_files = ZipFile(audio_zip_file, 'r')
 
-    def get_file_paths(self, temp_dir):
-        """Return file paths from temp_dir checking subdirectories"""
-        file_paths = []
-        for root, directories, files in os.walk(temp_dir):
+        with tempfile.TemporaryDirectory() as temp_dir:
+            audio_files.extractall(temp_dir)
+
+            # let's create dict file_paths[surah][ayah] = file_path
+            file_paths = self.get_file_paths_dict(temp_dir=temp_dir)
+            # let's create Recitations from segments and file_paths
+            for surah_number, ayahs in segments_dict.items():
+                for ayah_number, segments in ayahs.items():
+                    with open(file_paths[surah_number][ayah_number], 'rb')\
+                            as audio_file:
+
+                        file_ayah = File(audio_file)
+                        ayah = Ayah.objects.get(
+                            number=ayah_number,
+                            surah=surah_number,
+                        )
+
+                        Recitation.objects.get_or_create(
+                            ayah=ayah,
+                            reciter=obj,
+                            segments=segments,
+                            audio=file_ayah
+                        )
+
+    def get_file_paths_dict(self, temp_dir):
+        """
+        Process audio zip file uploaded and create dict
+        :return: file_paths[surah][ayah] = file_path
+        """
+        file_paths = defaultdict(lambda: defaultdict(int))
+
+        for root, _directories, files in os.walk(temp_dir):
             for filename in files:
-                filepath = os.path.join(root, filename)
-                file_paths.append(filepath)
+                if filename.endswith('.mp3'):
+                    filepath = os.path.join(root, filename)
+
+                    # extract surah and ayah number from filename
+                    surah_str = filename[:3]
+                    ayah_str = filename[3:6]
+                    try:
+                        surah_number = int(surah_str)
+                        ayah_number = int(ayah_str)
+                    except ValueError:
+                        print(f"Not normal format mp3 file {filename}")
+                    file_paths[surah_number][ayah_number] = filepath
 
         return file_paths
 
-    def get_ayah_object_from_row(self, row):
-        """Getting Ayah object for Recitation from csv file row"""
-        ayah_id = row.get('ayat')
-        surah_id = row.get('sura')
-
-        ayah = Ayah.objects.get(
-            number=ayah_id,
-            surah=surah_id,
-        )
-        return ayah
-
-    def get_file_audio_path(self, file_paths, ayah):
-        """Search in uploaded files for corresponding ayah"""
+    def get_segments_dict(self, request, form):
```

Comment:
I would make this function more specific. Instead of giving `request` and `form` I would give only those parts of these objects that you are using here.